.github/workflows/codeql-analysis.yml

```diff
@@ -20,6 +20,9 @@ on:
   schedule:
     - cron: '29 6 * * 6'
 
+permissions:  
+  contents: read
```

Comment:
(for other reviewers) see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token